axon-server-connector/src/main/java/org/axonframework/axonserver/connector/query/AxonServerQueryBus.java

```diff
@@ -352,11 +351,14 @@ public void onNext(QueryProviderInbound inboundRequest) {
 
                     @Override
                     public void onError(Throwable throwable) {
+                        logger.warn("Received error from server: {}", throwable.getMessage());
                         outboundStreamObserver = null;
+                        resubscribe();
```

Comment:
Do not resubscribe when error is StatusRuntimeException with Error code UNAVAILABLE. In this case the AxonServer is gone, which is handled by the PlatformConnectionManager.